cloudpickle/cloudpickle.py

```diff
@@ -69,6 +69,43 @@
     from io import BytesIO as StringIO
     PY3 = True
 
+
+try:
+    from ctypes import pythonapi, py_object, c_int, PYFUNCTYPE
+except ImportError:
+    supports_recursive_closure = False
+
+    def compress_closure(closure):
+        return closure
+
+    def decompress_closure(compressed_closure):
+        return compressed_closure
+
+    def fill_cells(cells, values):
+        pass
+else:
+    supports_recursive_closure = True
+
+    def compress_closure(closure):
+        return len(closure) if closure is not None else -1
```

Comment:
I just pushed a test that shows the problem. Using 0 causes the deserialized function to have a `__closure__` of `()` instead of None which is different behavior.